roadmap/2020/MLOpsRoadmap2020.md

```diff
@@ -142,7 +149,7 @@ MLOps must be framework-agnostic. There are a plethora of different Machine Lear
 
 MLOps must be platform and infrastructure agnostic. Adoption of MLOps is predicated upon being able to operate using this approach within the constraints of previously defined corporate infrastructure. It should be presumed unlikely that MLOps alone is a sufficient driver to motivate fundamental change to infrastructure.
 
-MLOps should strive to be agnostic to target hardware. Models may expect to be trained or executed on CPU, GPU, TPU, dedicated ASICs or custom neuro-morphic silicon. It should be possible to use MLOps across the widest possible range of target hardware. In some scenarios, it may also be desirable to facilitate ‘cross-compilation’ where models are trained on one class of hardware to optimise for training speed and are deployed onto a different class of hardware to optimise for cost/performance.
+MLOps should strive to be agnostic to hardware. Models may expect to be trained or executed on CPU, GPU, TPU, dedicated ASICs or custom neuro-morphic silicon. It should be possible to use MLOps across the widest possible range of hardware. It is also desirable to have the ability to train once but run anywhere, that is, models are trained on specific hardware to minimize training time, whereas they are optimized for different target devices based on cost or performance considerations.
```

Comment:
I feel this dilutes the specific meaning of the original statement. It will be very hard to build truly agnostic solutions on the training side due to the need for specific software support for given physical compute resources. Similarly, there are numerous compatibility challenges on target hardware. In the Cloud, train on GPU, run on CPU or similar may be generally achievable with a lot of work at a framework level but is challenging today. Train on GPU, run on ASIC is commercially very important and represents the most extreme case of 'cross-compilation'.

Reply:
I am not sure I understand the meaning of the original statement and how it's markedly different from the current. Perhaps you can clarify.

It still says that it should be agnostic of the target hardware. The reason I removed 'target' from the first sentence is because (as it already said), models can be trained on CPU/GPU/TPU/what have you. MLOps needs to encompass both training _and_ target hardware. 

Perhaps we cannot expect MLOps to be _fully_ agnostic of the training hardware, since supported hardware for training is framework-dependent, but that is true of target hardware, too. The overwhelming majority of use cases will involve training on CPUs (maybe GPUs) and deploying in the cloud/on premise onto CPUs (Model -> Docker -> Kubernetes appears to be the go-to way in the industry at the moment).

Reply:
Ok, this suggests to me that more clarity is required in a more specific manner so I suggest the attached expansion to resolve this.

Reply:
it seems like a good aim, but that train once/specialised seems perhaps out of the scope of this?

Reply:
With IoT devices becoming more prevalent and the industry embracing that development too, I think there may be a need in the next 5-10 years to support train once / run (almost) everywhere. I agree it's out of scope for the 2020 roadmap, but perhaps not if we look beyond the current year.